modules/contact/src/auxkernels/MortarFrictionalPressureVectorAux.C

```diff
@@ -0,0 +1,124 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "MortarFrictionalPressureVectorAux.h"
+#include "SystemBase.h"
+
+registerMooseObject("ContactApp", MortarFrictionalPressureVectorAux);
+
+InputParameters
+MortarFrictionalPressureVectorAux::validParams()
+{
+  InputParameters params = AuxKernel::validParams();
+  params.set<ExecFlagEnum>("execute_on") = EXEC_NONLINEAR;
+
+  params.addClassDescription("This class creates an auxiliary vector for outputting the mortar "
+                             "frictional pressure vector.");
+  params.addRequiredCoupledVar(
+      "tangent_one",
+      "First tangent vector Lagrange's multiplier for computing the mortar "
+      "frictional pressure vector.");
+  params.addRequiredCoupledVar(
+      "tangent_two",
+      "Second tangent vector Lagrange's multiplier for computing the mortar "
+      "frictional pressure vector.");
+  params.addRequiredParam<unsigned int>("component",
+                                        "Cartesian component of frictional tangent vector");
+
+  params.addRequiredParam<BoundaryName>("primary_boundary",
+                                        "The name of the primary boundary sideset.");
+  params.addRequiredParam<BoundaryName>("secondary_boundary",
+                                        "The name of the secondary boundary sideset.");
+
+  return params;
+}
+
+MortarFrictionalPressureVectorAux::MortarFrictionalPressureVectorAux(const InputParameters & params)
+  : AuxKernel(params),
+    _tangent_one(&coupledValueLower("tangent_one")),
+    _tangent_two(&coupledValueLower("tangent_two")),
+    _fe_problem(*params.get<FEProblemBase *>("_fe_problem_base")),
+    _primary_id(_fe_problem.mesh().getBoundaryID(getParam<BoundaryName>("primary_boundary"))),
+    _secondary_id(_fe_problem.mesh().getBoundaryID(getParam<BoundaryName>("secondary_boundary"))),
+    _component(getParam<unsigned int>("component"))
+{
+  // Only consider nodal quantities
+  if (!isNodal())
+    mooseError(
+        "MortarFrictionalPressureVector auxiliary kernel can only be used with nodal kernels.");
+
+  // Kernel need to be boundary restricted
+  if (!this->_bnd)
+    paramError("boundary",
+               "MortarFrictionalPressureVector auxiliary kernel must be restricted to a boundary.");
+
+  // Get mortar interfaces
+  const auto & displaced_mortar_interfaces = _fe_problem.getMortarInterfaces(/*displaced=*/true);
```

Comment:
Hmm could we set the `use_displaced_mesh` parameter to true in `validParams` and then do a `paramError` if it's false in the constructor?